### ISSUE TITLE
Broken FP support for armv6-m

### ISSUE THREAD

**Issue by nashif** [zephyriot/zep-jira14#1370]
**_Reported by Marcus Shawcroft:_**

On master,

Build attached main.c with:

CONFIG_NEWLIB_LIBC=y

When built for frdm_k64f (cortex-m4, armv7em) , binary behaves as expected:
make BOARD=frdm_k64f
Hello World! 3.000000
Hello World! 3.200000
Hello World! 3.400000
Hello World! 3.600000
Hello World! 3.800000
Hello World! 4.000000
Hello World! 4.200000
Hello World! 4.400000
Hello World! 4.600000
Hello World! 4.800000

When built for bbc_microbit (cortex-m0 armv6-m), binary outputs junk:
Hello World! ?." ?
Hello World! ?." ?
Hello World! ?." ?
Hello World! ?." ?
...

Suspect we still have issues with multlibs....

The presence / absence of  CONFIG_NEWLIB_LIBC_FLOAT_PRINTF appears to have no effect on either build which may be an issue in its own right since I expected the absence of this option to include a newlib nano build with no float printf support....??                                                                                                                             │  

(Imported from Jira ZEP-1503)

**Comment by nashif** [comment]
**_by Marcus Shawcroft:_**

* [main.c](https://s3.amazonaws.com/logs.zephyrproject.org/jira/GH-1370/main.c) (size: 163)

**Comment by nashif** [comment]
**_by Sharron LIU:_**

_Maureen Helm_ , can you take a look first? It does not seem to be newlib issue, since FRDM works.

**Comment by nashif** [comment]
**_by Maureen Helm:_**

Hi _Marcus Shawcroft_ , I tested this on frdm_k64f and frdm_kw41z (cortex m0+), and both work properly. The trick was to set CONFIG_STDOUT_CONSOLE=y.

Initially I tried your your main.c and got no printf output at all. The fxos8700 sample uses float printf and works fine, so then I started comparing configs, and found the STDOUT_CONSOLE was different. What I don't understand is, if this was the issue, then why did you get anything out to the console?
